File: src/extract_census.py
import json
import requests
from google.cloud import storage
import functions_framework

@functions_framework.http
def extract_data(request):
    client = storage.Client()
    bucket = client.bucket('mjumbewu_musa_509_raw_data')

    url = 'https://api.census.gov/data/2020/dec/pl?get=NAME,GEO_ID,P1_001N&for=block%20group:*&in=state:42%20county:*'
    resp = requests.get(url)
    data = resp.json()

    json_data = json.dumps(data)
    blob = bucket.blob('census/census_population_2020.json')
    blob.upload_from_string(json_data, content_type='application/json')

    return 'OK'

File: src/extract_census.mjs
import fetch from 'node-fetch';
import storage from '@google-cloud/storage';
import functions from '@google-cloud/functions-framework';

functions.http('extract_data', async (req, res) => {
  const client = new storage.Storage();
  const bucket = client.bucket('mjumbewu_musa_509_raw_data');

  const url = 'https://api.census.gov/data/2020/dec/pl?get=NAME,GEO_ID,P1_001N&for=block%20group:*&in=state:42%20county:*';
  const resp = await fetch(url);
  const data = await resp.json();

  const jsonData = JSON.stringify(data);
  const blob = bucket.file('census/census_population_2020.json');
  await blob.save(jsonData, { resumable: false });

  res.status(200).send('OK');
});
